src/lib/navigation/navigation-mode-factory.provider.ts

```diff
@@ -0,0 +1,78 @@
+import {FreeNavigationMode} from './free-navigation-mode';
+import {NavigationMode} from './navigation-mode.interface';
+import {SemiStrictNavigationMode} from './semi-strict-navigation-mode';
+import {StrictNavigationMode} from './strict-navigation-mode';
+import {WizardComponent} from '../components/wizard.component';
+import {NavigationModeInput} from './navigation-mode-input.interface';
+
+/**
+ * A factory used to create [[NavigationMode]] instances
+ */
+export class BaseNavigationModeFactory {
```

Comment:
`... implements NavigationModeFactory` is missing

Reply:
I must have missed this file when reviewing my changes. Fixed in 72bae52.